WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemDataSourceTest.kt

```diff
@@ -1,172 +1,163 @@
-
-        val identifiers = dataSource.getItemIdentifiers(mock(), remoteItemIds = emptyList(), isListFullyFetched = true)
-        assertThat(identifiers.size).isEqualTo(0)
+        testSectionedItemIdentifiers(isListFullyFetched = true, excludeListTypes = PostListType.values().toList())
     }
 
-    private fun makePosts(): List<PostModel> {
-        val posts = mutableListOf<PostModel>()
-        val statuses = listOf("draft", "publish", "pending", "trash", "future", "private")
-        statuses.forEachIndexed { i, status ->
-            val post = PostModel()
-            post.setId(i + 1)
-            post.setStatus(status)
-            posts.add(post)
+    /**
+     * This helper function sets up and tests sectioned identifiers for a test.
+     *
+     * 1. First create postSummaries & posts to be used for remote and local posts respectively.
+     * 2. Use these lists as mocks for PostStore getters.
+     * 3. Use the actual data source to get the identifiers which will use the mocked data.
+     * 4. Create a reversed mutable section headers list while respecting the [excludeListTypes] parameter:
+     * 5. For each identifier:
+     * * If it's a post, find its status by using the mocked data and then verify that it belongs to
+     * the current section.
+     * * If it's an end list indicator, check if we were expecting one.
+     * * If it's a section header, check that it's the current section header and then pop the current section header
+     * from the mutable list, so the next section becomes available.
+     * 6. Finally, verify that all expected sections are accounted for.
+     */
+    private fun testSectionedItemIdentifiers(
+        isListFullyFetched: Boolean,
+        excludeListTypes: List<PostListType> = emptyList()
+    ) {
+        val postSummaries = createPostSummaries(excludeListTypes)
+        val posts = createLocalPosts(excludeListTypes)
+        setupPostStoreMocks(postSummaries, posts)
+
+        val identifiers = dataSource.getItemIdentifiers(
+                mock(),
+                remoteItemIds = emptyList(), // doesn't matter since we mocked the PostStore getters
+                isListFullyFetched = isListFullyFetched
+        )
+        val getRemotePostStatus = { remotePostId: RemotePostId ->
+            postSummaries.find { it.remoteId == remotePostId.id.value }!!.status
         }
-
-        return posts
+        val getLocalPostStatus = { localPostId: LocalPostId ->
+            PostStatus.fromPost(posts.find { it.id == localPostId.id.value }!!)
+        }
+        val mutableSectionHeaders = SECTION_HEADERS.filter { !excludeListTypes.contains(it.type) }.reversed()
+                .toMutableList()
+        identifiers.reversed().forEach { identifier ->
+            val currentSection = mutableSectionHeaders.first()
+            when (identifier) {
+                is LocalPostId -> {
+                    assert(PostListType.fromPostStatus(getLocalPostStatus(identifier)) == currentSection.type)
```

Comment:
I don't actually know what the difference is, but switched to `assertTrue` of junit in c91f76d4b53b546cb692ffa2233659d8b2a032b7. Let me know if that works for you.

Reply:
A couple of differences
-  `assert` works only when you start the JVM with a flag -> it always passes otherwise. (I thought the AS is smart enough to turn it on when running unit-tests. But it actually isn't ->  the tests were actually always passing no matter the value)
- `assertThat` automatically tells you what was the expected and actual values when the test fails (as oppose to `assertTrue` and `assert`) (more differences [here](https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/))
- consistency:) - we use `assertThat` in most unit tests (`assertTrue` is used in over 100 places as well, so I'm fine with that.)

`assert` is sometimes for example used in debug builds to follow the fail fast principle.

Reply:
@malinajirka Thank you for elaborating on this. I've made the change to `assertThat` in f0906511c5d50c6ac2ac0b2347c89484d4a63c8f. I should have done that instead of `assertTrue` in the first place, sorry about that!

Reply:
No worries at all, thanks! ;)